Add Video Calling/iOSVideo/CallKitHelper.swift

```diff
@@ -0,0 +1,553 @@
+                action.fail()
+                return
+            }
+            
+            do {
+                if action.isOnHold {
+                    try await stopAudio(call: activeCall)
+                    try await activeCall.hold()
+                } else {
+                    // Dont resume the audio here, have to to wait for `didActivateAudioSession`
+                    try await activeCall.resume()
+                }
+                action.fulfill()
+            } catch {
+                action.fail()
+            }
+        }
+    }
+    
+    func provider(_ provider: CXProvider, perform action: CXSetMutedCallAction) {
+        Task {
+            guard let activeCall = await self.callKitHelper.getActiveCall(callId: action.callUUID.uuidString) else {
+                action.fail()
+                return
+            }
+
+            do {
+                if action.isMuted {
+                    try await activeCall.mute()
+                } else {
+                    try await activeCall.unmute()
+                }
+                action.fulfill()
+            } catch {
+                action.fail()
+            }
+        }
+    }
+
+    func provider(_ provider: CXProvider, perform action: CXAnswerCallAction) {
+        Task {
+            let error = configureAudioSession()
+            
+            // this can be nil and its ok because this can also directly come from CallKit
+            let outInCallInfo = await callKitHelper.getOutInCallInfo(transactionId: action.uuid)
+            
+            if error != nil {
+                action.fail()
+                outInCallInfo?.completionHandler(nil, error)
+                return
+            }
+
+            let completionBlock : ((Call?, Error?) -> Void) = { (call, error) in
+                
+                if error == nil {
+                    action.fulfill()
+                    Task {
+                        await self.callKitHelper.addActiveCall(callId: action.callUUID.uuidString,
+                                                               call: call!)
+                    }
+                } else {
+                    action.fail()
+                }
+                outInCallInfo?.completionHandler(call, error)
+                Task {
+                    await self.callKitHelper.removeOutInCallInfo(transactionId: action.uuid)
+                    await self.callKitHelper.removeIncomingCall(callId: action.callUUID.uuidString)
+                }
+            }
+            
+            let acceptCallOptions = outInCallInfo?.options as? AcceptCallOptions
+
+            if let incomingCall = await callKitHelper.getIncomingCall(callId: action.callUUID) {
+                incomingCall.accept(options: acceptCallOptions ?? AcceptCallOptions(), completionHandler: completionBlock)
+                return
+            }
+            
+            let dispatchSemaphore = await self.callKitHelper.setAndGetSemaphore()
```

Comment:
Is there a way to avoid using a semaphore and instead use task groups or async / await?

Reply:
Async/Await is not really possible here because the event that raises the incoming call comes from a different process than the CallKit process, so I am not seeing any other way than semaphore here.